File: components/ntp_snippets/remote/test_utils.cc
FASTVC-BENCH-CORPUS:chromium-main-v1-943b94ae-1c74-4335-94fa-ceb4d277cea8
// Copyright 2016 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "components/ntp_snippets/remote/test_utils.h"

#include <memory>

namespace ntp_snippets {

namespace test {

RemoteSuggestionsTestUtils::RemoteSuggestionsTestUtils()
    : pref_service_(std::make_unique<TestingPrefServiceSyncable>()) {}

RemoteSuggestionsTestUtils::~RemoteSuggestionsTestUtils() = default;

}  // namespace test

}  // namespace ntp_snippets

File: chrome/test/data/web_apps/isolated_app/test_utils.js
FASTVC-BENCH-CORPUS:chromium-main-v1-943b94ae-1c74-4335-94fa-ceb4d277cea8
// Copyright 2021 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

function addManifestLinkTag(optionalCustomUrl) {
  const url = new URL(window.location.href);
  let manifestUrl = url.searchParams.get('manifest');
  if (!manifestUrl) {
    manifestUrl = optionalCustomUrl || 'basic.json';
  }

  var linkTag = document.createElement("link");
  linkTag.id = "manifest";
  linkTag.rel = "manifest";
  linkTag.href = `./${manifestUrl}`;
  document.head.append(linkTag);
}

function startWorker(worker, options) {
  navigator.serviceWorker.register(worker, options);
}
